website/blog/2020-02-06-react-navigation-5.0.md

```diff
@@ -0,0 +1,172 @@
+
+```js
+navigation.setOptions({
+  headerRight: () => (
+    <DoneButton
+      onPress={async () => {
+        await saveNote();
+        navigation.replace('Notes');
+      }}
+    />
+  ),
+})
+```
+
+It can be used for things like adding a button in the header which needs to interact with the screen state.
+
+### New theming API
+
+In React Navigation, we had basic theming support where you could specify whether to use a light or dark theme. It wasn't easy to customize the colors used by the built-in components such as header, tab bar etc. without extra code or repetition.
+
+Now, we have revamped the [theme system](https://reactnavigation.org/docs/themes.html) for easier customization. It is possible to provide a theme object with your desired colors for background, accent color etc. and it will automatically change the colors of all navigators without any extra code.
+
+```js
+const MyTheme = {
+  dark: false,
+  colors: {
+    primary: 'rgb(255, 45, 85)',
+    background: 'rgb(242, 242, 242)',
+    card: 'rgb(255, 255, 255)',
+    text: 'rgb(28, 28, 30)',
+    border: 'rgb(199, 199, 204)',
+  },
+};
+```
+
+### First-class types with TypeScript
+
+The new version has been written from the ground-up with TypeScript. We now get first class autocompletion and type-checking.
+
+![TypeScript in action](assets/announcing-5.0/typescript.gif)
+
+We also have JSDoc for the built-in methods and options, so you get their description directly in your editor. See [our typescript documentation](https://reactnavigation.org/docs/typescript.html) for more details on how to use it.
+
+### Redux DevTools integration
+
+If you use [React Native Debugger](https://github.com/jhen0409/react-native-debugger) or [Redux Devtools Extension](https://github.com/zalmoxisus/redux-devtools-extension), you can see navigation actions in the devtools along with the current navigation state. It also supports time-travel debugging!
+
+![Redux Devtools in action](assets/announcing-5.0/redux-devtools.gif)
+
+You don't need to use Redux in your apps for this to work and it works without any extra setup!
+
+### Native Stack Navigator
+
+Traditionally, we have written our navigators in JavaScript for greater customizability. It fits a lot of use cases, but sometimes you want the exact native feel and the performance of native navigation. Now, we have added a new native stack navigator that uses native navigation primitives for navigation using the [`react-native-screens`](https://github.com/kmagiera/react-native-screens) library. Under the hood it just uses naitive components which might be obvious choice for native developement and might be good pick in the most cases. 
```

Comment:
2 typos in last sentence. Should be:

_Under the hood it just uses **native** components which might be obvious choice for native **development** and might be good pick in the most cases._